### PR TITLE
Update string widget to use IME_ACTION_NEXT for short appearance

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
#2281 

Screenshots after the change: 
Portrait
WhatsApp Image 2020-11-10 at 3 12 50 PM (1)
Landscape
WhatsApp Image 2020-11-10 at 3 12 50 PM
Tablet
Screenshot 2020-11-10 at 3 37 30 PM

### CODE REVIEW

File: app/res/values/dimens.xml [hunk 1]
```diff
@@ -128,7 +128,7 @@
 
     <!--Big Layout Button Dimensions-->
     <!-- Should be font medium SP x 4, roughly -->
-    <dimen name="rectangle_button_height">60sp</dimen>
+    <dimen name="rectangle_button_height">40dp</dimen>
```
Comment (shubham1g5) [inline app/res/values/dimens.xml:131]: nice job using `dp` for this.

File: app/src/org/commcare/views/widgets/StringWidget.java [hunk 2]
```diff
@@ -238,6 +235,13 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
     }
 
     public void setLastQuestion(boolean isLast) {
+        if (SHORT_APPEARANCE.equalsIgnoreCase(mPrompt.getAppearanceHint()) || isInCompactMode()) {
```
Comment (shubham1g5) [inline app/src/org/commcare/views/widgets/StringWidget.java:238]: nit: would be nice to abstract this logic in a method

**Review by shubham1g5 — APPROVED** [review]

**Comment by shubham1g5** [comment]
@ShivamPokhriyal  we should also update these docs sighting these changes from 2.51 onwards - 
https://confluence.dimagi.com/pages/viewpage.action?spaceKey=ccinternal&title=Single+Line+Numeric+Entry
https://confluence.dimagi.com/display/commcarepublic/Advanced+CommCare+Android+Formatting#AdvancedCommCareAndroidFormatting-SingleLineNumericEntry and also add `short` appearance behaviour in this doc.

**Comment by ShivamPokhriyal** [comment]
Updated both the docs https://confluence.dimagi.com/display/commcarepublic/Advanced+CommCare+Android+Formatting#AdvancedCommCareAndroidFormatting-ShortTextEntryWidget.

**Comment by dimagimon** [comment]
[32% (0.0%) vs master 32%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462129/)

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**